### PR TITLE
[eventgrid] add tests for samples

### PR DESCRIPTION
**Pull request by rakshith91**
Samples are triggered in the live run
Example: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=679428&view=results

Fixes https://github.com/Azure/azure-sdk-for-python/issues/14322

### CODE REVIEW

File: sdk/eventgrid/azure-eventgrid/samples/champion_scenarios/cs3_consume_system_events.py
Comment (rakshith91) [inline sdk/eventgrid/azure-eventgrid/samples/champion_scenarios/cs3_consume_system_events.py:18]:
```suggestion
path = os.path.abspath(os.path.join(os.path.abspath(__file__), "..", "./cs3_system_events.json"))
```

Comment (rakshith91) [inline sdk/eventgrid/azure-eventgrid/samples/champion_scenarios/cs3_consume_system_events.py]:
```suggestion
path = os.path.abspath(os.path.join(os.path.abspath(__file__), "..", "./cs3_consume_system_event.json"))
```

File: sdk/eventgrid/azure-eventgrid/samples/champion_scenarios/cs3_consume_system_events.py
Comment (rakshith91) [inline sdk/eventgrid/azure-eventgrid/samples/champion_scenarios/cs3_consume_system_events.py:18]:
```suggestion
path = os.path.abspath(os.path.join(os.path.abspath(__file__), "..", "./cs3_system_event.json"))
```

File: scripts/devops_tasks/test_run_samples.py
Comment (rakshith91) [inline scripts/devops_tasks/test_run_samples.py:29]: Will be fixed in this https://github.com/Azure/azure-sdk-for-python/issues/16042

File: scripts/devops_tasks/test_run_samples.py
Comment (rakshith91) [inline scripts/devops_tasks/test_run_samples.py:28]: it's not practical to test these these 4 samples - they have a dependency on storage, servicebus, EH, azure functions.
it's creating a lot of maintainance in tests.yml and in adding more dependcies to dev_requirements. Also, it's not simple to deploy azure functions on CI sample run and moreover, we do have consume samples which mimic the data we receive.

File: scripts/devops_tasks/test_run_samples.py
Comment (rakshith91) [inline scripts/devops_tasks/test_run_samples.py:29]:
```suggestion
        "cs2_publish_custom_events_to_a_domain_topic.py"]
```

File: scripts/devops_tasks/test_run_samples.py
Comment (yunhaoling) [reply scripts/devops_tasks/test_run_samples.py:28]: you're farsighted, thanks for calling it out!

but I'm afraid that in the long term we want to have tests against those integration samples. Because it helps us verify if there're any changes in EH/SB/Storage SDK and we'll need to upgrade these EG samples accordingly.

I'm not sure how other SDKs is doing, but I think it would be nice to have tests for integration samples.
I didn't intent to block the PR, we could open separate issue and PRs when we have time.

thoughts? @rakshith91 @lmazuel

File: scripts/devops_tasks/test_run_samples.py
Comment (yunhaoling) [inline scripts/devops_tasks/test_run_samples.py:25]: dumb question for my own learning: why here is a `__init__.py`, I don't see there is a `__init__.py` inside the sample folder?

Comment (rakshith91) [reply scripts/devops_tasks/test_run_samples.py:25]: the __init__.py is from the azurefunctions appi was talking about in my previous comment
https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/eventgrid/azure-eventgrid/samples/consume_samples/e2e_samples/functionsapp

File: scripts/devops_tasks/test_run_samples.py
Comment (rakshith91) [reply scripts/devops_tasks/test_run_samples.py:28]: that's a good point - i agree that integration tests can be useful in the long run when we eventually have some breaking changes - what i can do is add these tests once we move the test framework to use the arm template in the march milestone - that way, we have less maintenance on tests.yml

Comment (yunhaoling) [reply scripts/devops_tasks/test_run_samples.py:28]: sounds good to me!

File: scripts/devops_tasks/test_run_samples.py
Comment (yunhaoling) [reply scripts/devops_tasks/test_run_samples.py:25]: oh I see, then this is for EG only, make sense to me

File: scripts/devops_tasks/test_run_samples.py
Comment (rakshith91) [reply scripts/devops_tasks/test_run_samples.py:28]: https://github.com/Azure/azure-sdk-for-python/issues/16057

**Review by kristapratico — APPROVED** [review]
LGTM in terms of running samples

**Review by yunhaoling — APPROVED** [review]
LGTM, do we need to kick off a livetest run against this PR before merging?

**Comment by rakshith91** [comment]
> do we need to kick off a livetest run against this PR before merging?

already did and shared the link on the pr description

**Pull request merged**
